cmd/controller/sparsefilegenerator.go

```diff
@@ -0,0 +1,180 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/pkg/util"
+
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+)
+
+/*
+Sparse File help simulate disk objects that can be used for testing and
+proto typing solutions built using node-disk-manager(NDM).  Sparse files
+will be created if NDM is provided with the location where sparse files
+should be located.
+
+On Startup, if a sparse directory (SPARSE_FILE_DIR) is specified as a
+environment variable, a Sparse file with specified size (SPARSE_FILE_SIZE) will
+be created and an associated Disk CR will be added to Kubernetes. By default
+only one sparse file will be created which can be changed by passing the desired
+number of sparse files required via the environment variable SPARSE_FILE_COUNT.
+
+On Shutdown, the status of the sparse file Disk CR will be marked as Unknown.
+*/
+
+const (
+	ENV_SPARSE_FILE_DIR   = "SPARSE_FILE_DIR"
+	ENV_SPARSE_FILE_SIZE  = "SPARSE_FILE_SIZE"
+	ENV_SPARSE_FILE_COUNT = "SPARSE_FILE_COUNT"
+
+	SPARSE_FILE_NAME          = "ndm-sparse.img"
+	SPARSE_FILE_DEFAULT_SIZE  = "1073741824"
+	SPARSE_FILE_DEFAULT_COUNT = "1"
+
+	SPARSE_DISKTYPE    = "sparse"
+	SPARSE_DISK_PREFIX = "sparse-"
+)
+
+// GetSparseFileDir returns the full path to the sparse
+//  file directory on the node.
+func GetSparseFileDir() string {
+
+	sparseFileDir := os.Getenv(ENV_SPARSE_FILE_DIR)
+
+	if len(sparseFileDir) < 1 {
+		return ""
+	}
+
+	info, err := os.Stat(sparseFileDir)
+	if os.IsNotExist(err) || !info.Mode().IsDir() {
+		glog.Info("Specified directory doesnt exist:  ", sparseFileDir)
+		return ""
+	}
+
+	return sparseFileDir
+}
+
+// GetSparseFileCount returns the number of sparse files to be
+//  created by NDM. Returns 0, if invalid count is specified.
+func GetSparseFileCount() int {
+
+	sparseFileCountStr := os.Getenv(ENV_SPARSE_FILE_COUNT)
+
+	if len(sparseFileCountStr) < 1 {
+		sparseFileCountStr = SPARSE_FILE_DEFAULT_COUNT
+	}
+
+	sparseFileCount, econv := strconv.Atoi(sparseFileCountStr)
+	if econv != nil {
+		glog.Info("Error converting sparse file count:  ", sparseFileCountStr)
+		return 0
+	}
+
+	return sparseFileCount
+}
+
+// GetSparseFileSize returns the size of the sparse file to be
+//  created by NDM. Returns 0, if invalid size is specified.
+func GetSparseFileSize() int64 {
+
+	sparseFileSizeStr := os.Getenv(ENV_SPARSE_FILE_SIZE)
+	if len(sparseFileSizeStr) < 1 {
+		sparseFileSizeStr = SPARSE_FILE_DEFAULT_SIZE
+	}
+
+	sparseFileSize, econv := strconv.ParseInt(sparseFileSizeStr, 10, 64)
+	if econv != nil {
+		glog.Info("Error converting sparse file size:  ", sparseFileSizeStr)
+		return 0
+	}
```

Comment:
Should it bail out if the size is less than a expected minimum value ?

Reply:
Could do this @avishnu -- can we assume 1G as minimum or probably go with user defined value for now and add the minimum value check based on usage.

Reply:
yes.. there are some minimums.. can't be less than 64MB.. so, lets make the minimum as 512MB.. lets take it as 1G as 1G is not too much to ask

Reply:
ok. I will go with setting the minimum as 1G.